From 42ce853af92e8cae55bf93275409d31e41dd43f0 Mon Sep 17 00:00:00 2001
From: Sr-0w <Sr-0w@users.noreply.github.com>
Date: Mon, 10 Aug 2026 15:04:00 +0200
Subject: [PATCH] arm64: dts: qcom: select Semco IMX586 lens on hotdog

The OxygenOS camera data contains two Semco IMX586 module variants. The
AK7374 variant does not acknowledge at 0x0c on the HD1913 bring-up unit,
while the alternate Semco module uses an LC898217XC at 0x72. Describe that
physical variant and retain the stock 0..400 calibrated range.

---
 arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts b/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts
index 8ea2634e3b63..f0b5794f1df5 100644
--- a/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts
+++ b/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts
@@ -958,12 +958,14 @@ s5k3m5_ep: endpoint {
 	/* The stock AK7374 actuator configuration requires CCI FAST mode. */
 	clock-frequency = <400000>;
 
-	imx586_vcm: lens@c {
-		compatible = "asahi-kasei,ak7374";
-		reg = <0x0c>;
+	imx586_vcm: lens@72 {
+		compatible = "onnn,lc898217xc";
+		reg = <0x72>;
 
 		vdd-supply = <&vreg_cam0_vaf>;
 		vio-supply = <&vreg_l1c_1p8>;
+		onnn,max-focus-position = <400>;
+		onnn,default-focus-position = <400>;
 	};
 
 	/* Main 48 MP rear camera, identified on hardware as Sony IMX586. */
-- 
2.54.0
